webapp-django/crashstats/crashstats/tests/test_views.py

```diff
@@ -1549,6 +1550,44 @@ def mocked_supersearch_get(**params):
         ok_('Our database is experiencing troubles' in response.content)
         ok_('week of 2010-01-04 is ~40% lower' in response.content)
 
+    @mock.patch('requests.get')
+    def test_crashes_per_day_bad_argument_error(self, rget):
+        url = reverse('crashstats:crashes_per_day')
+
+        def mocked_adi_get(**options):
+            return {
+                'total': 0,
+                'hits': []
+            }
+
+        models.ADI.implementation().get.side_effect = mocked_adi_get
+
+        def mocked_product_build_types_get(**options):
+            return {
+                'hits': {
+                    'release': 0.1,
+                    'beta': 1.0,
+                }
+            }
+
+        models.ProductBuildTypes.implementation().get.side_effect = (
+            mocked_product_build_types_get
+        )
+
+        def mocked_supersearch_get(**params):
+            raise BadArgumentError('Someone is wrong on the Internet')
+
+        SuperSearchUnredacted.implementation().get.side_effect = (
+            mocked_supersearch_get
+        )
+
+        response = self.client.get(url, {
+            'p': 'WaterWolf',
+            'v': ['20.0', '19.0']
+        })
+        eq_(response.status_code, 400)
+        ok_('Someone is wrong on the Internet' in response.content)
```

Comment:
The point of this test is to capture `BadArgumentError` exceptions inside the implementation class. 

Back in the day when we had a middleware service, if there was anything wrong with the data the middleware server would respond with a `400 Bad Request` and we would sometimes let that bubble and sometimes re-wrap that. Then we did away with the middleware service and let django "talk directly" to the python code and now we could get `BadArgumentError` exceptions in the django view functions.